### PR TITLE
Fix multi-search with template doesn't return status code

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description

Currently the multi-search API does return the status code in each search response, but multi-search template API does not, these two APIs are very similar, we should make the response consistent, so this PR adds a status code field in each search response for the multi-search template API.

### Related Issues
https://github.com/opensearch-project/OpenSearch/issues/11133

### Check List
- [ ] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: modules/lang-mustache/src/main/java/org/opensearch/script/mustache/SearchTemplateResponse.java [hunk 1]
```diff
@@ -120,7 +120,10 @@ public static SearchTemplateResponse fromXContent(XContentParser parser) throws
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         if (hasResponse()) {
-            response.toXContent(builder, params);
+            builder.startObject();
```
Comment (reta) [inline modules/lang-mustache/src/main/java/org/opensearch/script/mustache/SearchTemplateResponse.java:123]: :+1:

**Comment by reta** [comment]
Thanks @gaobinlong , we would need API changes [1] (and may be documentation updates if we have any) [2].

 [1] https://github.com/opensearch-project/opensearch-api-specification
 [2] https://github.com/opensearch-project/documentation-website/

File: modules/lang-mustache/src/main/java/org/opensearch/script/mustache/SearchTemplateResponse.java (hunk 1 above)
Comment (reta) [reply modules/lang-mustache/src/main/java/org/opensearch/script/mustache/SearchTemplateResponse.java:123]: Hm ... Looking closely, we do actually change schema here. The `SearchResponse` was serialized to / from XContent as-is, but now we add this additional property `status` that we serialize along with it. I think this is fine but needs documentation / spec update :100%

**Review by reta — APPROVED** [review]

**Comment by gaobinlong** [comment]
> Thanks @gaobinlong , we would need API changes [1] (and may be documentation updates if we have any) [2].
> 
> [1] https://github.com/opensearch-project/opensearch-api-specification [2] https://github.com/opensearch-project/documentation-website/

Thanks, I've created a document PR:https://github.com/opensearch-project/documentation-website/pull/8522, and for the api-spec repo, I see now the msearch api and msearch template api uses same response schema: [MultiSearchResult](https://github.com/opensearch-project/opensearch-api-specification/blob/5dc65a531c5de10d423670a259c4b1905bd3407e/spec/namespaces/_core.yaml#L3074), so may no need to change it?

**Comment by reta** [comment]
> I see now the msearch api and msearch template api uses same response schema: [MultiSearchResult](https://github.com/opensearch-project/opensearch-api-specification/blob/5dc65a531c5de10d423670a259c4b1905bd3407e/spec/namespaces/_core.yaml#L3074), so may no need to change it?

Thanks @gaobinlong.  I just briefly looked at it, seems like `MultiSearchResult` has no `status` property? Or I am missing something?

**Comment by gaobinlong** [comment]
> > I see now the msearch api and msearch template api uses same response schema: [MultiSearchResult](https://github.com/opensearch-project/opensearch-api-specification/blob/5dc65a531c5de10d423670a259c4b1905bd3407e/spec/namespaces/_core.yaml#L3074), so may no need to change it?
> 
> Thanks @gaobinlong. I just briefly looked at it, seems like `MultiSearchResult` has no `status` property? Or I am missing something?

`status` exists in each search response, the definition is [here](https://github.com/opensearch-project/opensearch-api-specification/blob/5dc65a531c5de10d423670a259c4b1905bd3407e/spec/schemas/_core.msearch.yaml#L175), the whole response of the API is like this:
```
{
  "took": 8,
  "responses": [
    {
      "took": 7,
      "timed_out": false,
      "_shards": {...},
      "hits": {...},
      "status": 200
    },
   {
      "took": 7,
      "timed_out": false,
      "_shards": {...},
      "hits": {...},
      "status": 200
    }
  ]
}
```

**Comment by reta** [comment]
> > > I see now the msearch api and msearch template api uses same response schema: [MultiSearchResult](https://github.com/opensearch-project/opensearch-api-specification/blob/5dc65a531c5de10d423670a259c4b1905bd3407e/spec/namespaces/_core.yaml#L3074), so may no need to change it?
> > 
> > 
> > Thanks @gaobinlong. I just briefly looked at it, seems like `MultiSearchResult` has no `status` property? Or I am missing something?
> 
> `status` exists in each search response, the definition is [here](https://github.com/opensearch-project/opensearch-api-specification/blob/5dc65a531c5de10d423670a259c4b1905bd3407e/spec/schemas/_core.msearch.yaml#L175), the whole response of the API is like this:
> 
> ```
> {
>   "took": 8,
>   "responses": [
>     {
>       "took": 7,
>       "timed_out": false,
>       "_shards": {...},
>       "hits": {...},
>       "status": 200
>     },
>    {
>       "took": 7,
>       "timed_out": false,
>       "_shards": {...},
>       "hits": {...},
>       "status": 200
>     }
>   ]
> }
> ```

Perfrect, thank you!

**Review by reta — APPROVED** [review]

**Comment by dbwiddis** [comment]
Flaky:
 - https://github.com/opensearch-project/OpenSearch/issues/15944

**Comment by reta** [comment]
@gaobinlong could you please backport to `2.x` manually? thank you

**Pull request merged**